c10/core/impl/TorchDispatchModeTLS.h

```diff
@@ -6,20 +6,57 @@
 namespace c10 {
 namespace impl {
 
+enum class TorchDispatchModeKey : int8_t {
+  PROXY = 0,
+  FAKE = 1,
```

Comment:
This numbering seems reverse of what you want. If you add FUNCTIONALIZATION to this enum, it needs to come before PROXY, which is annoying because then you have to renumber everything. This is also inconsistent with DispatchKey ordering, where the last keys to run come first.

Reply:
It probably is a bad idea to explicitly number these mode keys too

Reply:
will do - I'll reverse them (I did indeed make FUNCTIONAL a zero in a later PR, I"ll flip)